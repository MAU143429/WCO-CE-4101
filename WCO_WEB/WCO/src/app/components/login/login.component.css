body{
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-color: white;
  background-image: url("../../../assets/images/bg.jpg");


}
.container{
  max-width: 700px;
  height: 650px;
  width: 100%;
  /**background-color: rgba(255, 255, 255, 0.295);*/
  background-color: white;
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}
.container .title{
  font-size: 35px;
  font-weight: 500;
  text-align: center;
  position: relative;
  margin-bottom: 2%;
}

.subtitle{
  font-size: 25px;
  font-weight: 500;
  text-align: center;
}

.container .title::before{
  content: "";
  position: absolute;
  bottom: 0;
  height: 3px;
  width: 210px;
  border-radius: 5px;
}
.content form .account-details{
  display: flex;
  flex-wrap: wrap;
  margin: 20px 0 12px 0;
}

form .account-details .input-box{
  margin-bottom: 40px;
  margin-left: 25%;
  width: calc(100% / 2);

}


form .input-box span.details{
  display: block;
  font-weight: 500;
  margin-bottom: 5px;
  text-align: center;
}



.account-details .input-box input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}

form .button{
  height: 45px;
  margin: 35px 0
}

 /* Aspecto del boton de crear partido */

 form .button input{
  height: 100%;
  width: 30%;
  margin-left: 35%;
  border-radius: 5px;
  border: none;
  color: #fff;
  font-size: 18px;
  font-weight: 500;
  letter-spacing: 1px;
  cursor: pointer;
  transition: all 0.3s ease;
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
}
 form .button input:hover{
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
  }
 @media(max-width: 584px){
 .container{
  max-width: 100%;
}
}

.social
{
  position: absolute;
  z-index: 10;
  bottom: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.social li
{
  list-style: none;
}
.social li a
{
  display: inline-block;
  margin-right: 20px;
  filter: invert(1);
  transform: scale(0.5);
  transition: 0.5s;
}
.social li a:hover
{
  transform: scale(0.5) translateY(-15px);
}

@media(max-width: 459px){
  .container .content{
    flex-direction: column;
  }
}
