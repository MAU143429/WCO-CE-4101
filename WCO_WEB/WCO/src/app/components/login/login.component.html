<p><!DOCTYPE html>
  <html lang="en" dir="ltr">
    <head>
      <meta charset="UTF-8">
      <!---Referencia al archivo css que contiene el estilo--->
      <link rel="stylesheet" href="login.component.css">
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
     </head>
  <body>
    <div class="container">
      <div class="title">Bienvenido a World Cup Online</div>
      <div class="content">
        <div class="subtitle">Iniciar Sesión</div>
        <br>
        <br>
        <form>
          <div class="account-details">
            <div class="input-box">
              <span class="details">Correo Electrónico<span style="color: rgb(245, 59, 59);">*</span> </span>
              <input type="email" minlength="5" maxlength="30" placeholder="Ingrese un correo electrónico.." required>
            </div>

            <div class="input-box">
              <span class="details">Contraseña<span style="color: rgb(245, 59, 59);">*</span> </span>
              <input type="password" minlength="6" maxlength="8" placeholder="Ingrese una contraseña aquí..." required>
            </div>
            <div class="input-box">
              <span >No tienes una cuenta aún? Regístrate <a routerLink="/register" style="color: rgb(4, 0, 255);">aquí</a></span>
            </div>
          </div>
          <div class="button">
            <input type="submit" value="Iniciar Sesión"routerLink="/home">
          </div>

        </form>
      </div>
    </div>
  </body>
  </html>
