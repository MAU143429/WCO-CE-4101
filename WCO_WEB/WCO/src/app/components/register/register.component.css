body{
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-image: url("../../../assets/images/bg.jpg");

}
.container{
  max-width: 1320px;
  width: 100%;
  background-color: white;
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}
.container .title{
  font-size: 30px;
  font-weight: 500;
  text-align: center;
  position: relative;
  margin-bottom: 2%;
}

.container .title::before{
  content: "";
  position: absolute;
  bottom: 0;
  height: 3px;
  width: 210px;
  border-radius: 5px;
  background: linear-gradient(135deg, #ffffff, #71aae4, #1277D9);
}
.content form .account-details{
  display: flex;
  flex-wrap: wrap;
  margin: 20px 0 12px 0;
}

form .account-details .input-box{
  margin-bottom: 40px;
  margin-left: 17%;
  width: calc(50% / 2);

}

form .account-details .input-box1{
  margin-bottom: 40px;
  margin-left: 37.5%;
  width: calc(50% / 2);

}

form .account-details .input-box2{
  margin-bottom: 40px;
  margin-left: 37.5%;
  width: calc(50% / 2);
  text-align: center;
}


form .input-box span.details{
  display: block;
  font-weight: 500;
  margin-bottom: 5px;
  text-align: center;
}

form .input-box1 span.details{
  display: block;
  font-weight: 500;
  margin-bottom: 5px;
  text-align: center;
}

form .input-box2 span.details{
  display: block;
  font-weight: 500;
  margin-bottom: 5px;
  text-align: center;
}


.account-details .input-box input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}
.account-details .input-box1 input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}

form .button{
  height: 45px;
  margin: 35px 0
}

 /* Aspecto del boton de crear partido */

 form .button input{
  height: 100%;
  width: 40%;
  margin-left: 30%;
  border-radius: 5px;
  border: none;
  color: #fff;
  font-size: 18px;
  font-weight: 500;
  letter-spacing: 1px;
  cursor: pointer;
  transition: all 0.3s ease;
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
}
 form .button input:hover{
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
  }
 @media(max-width: 584px){
 .container{
  max-width: 100%;
}
}
