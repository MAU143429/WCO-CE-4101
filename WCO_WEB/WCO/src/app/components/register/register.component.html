<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="register.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>
<body>
  <div class="container">
    <div class="title">Crear una cuenta</div>
    <div class="content">
      <form>
        <div class="account-details">

          <div class="input-box">
            <span class="details" >Nombre<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="text"  minlength="5" maxlength="30" placeholder="Ingrese un nombre aquí..." required>
          </div>
          <div class="input-box">
            <span class="details">Apellidos<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="text" minlength="5" maxlength="30" placeholder="Ingrese los apellidos aquí..." required>
          </div>

          <div class="input-box">
            <span class="details">Nickname<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="text" minlength="5" maxlength="30" placeholder="Ingrese un nickname..." required>
          </div>
          <div class="input-box">
            <span class="details">Correo Electrónico<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="email" minlength="5" maxlength="30" placeholder="Ingrese un correo electrónico.." required>
          </div>
          <div class="input-box">
            <span class="details">Contraseña<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="password" minlength="6" maxlength="8" placeholder="Ingrese una contraseña aquí..." required>
          </div>
          <div class="input-box">
            <span class="details"  >Fecha de nacimiento<span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="date"  required>
          </div>
          <div class="input-box1">
            <span class="details" >Equipo 1<span style="color: rgb(245, 59, 59);">*</span></span>
            <mbsc-select [data]="CData" [touchUi]="false"  placeholder="Seleccione su país de procedencia" ></mbsc-select>
          </div>

          <div class="input-box2">
            <span style="color: rgb(4, 0, 255);">Mostrar Términos y Condiciones</span><br>
            <span><input type="checkbox"> Aceptar Términos y Condiciones</span>
          </div>
          <div class="input-box2">
            <span>Ya tienes una cuenta? Ingresa <a routerLink="/login" style="color: rgb(4, 0, 255);">aquí</a></span>
          </div>
        </div>
        <div class="button">
          <input type="submit" value="Crear cuenta" routerLink="/login">
        </div>

      </form>
    </div>
  </div>
</body>
</html>
