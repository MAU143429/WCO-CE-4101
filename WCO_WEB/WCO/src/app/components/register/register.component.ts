import { Component, OnInit } from '@angular/core';
import { Countries } from '../../interface/countries';
import countriesData from '../../../assets/scripts/countries.json';
import { CreateAccount } from 'src/app/model/create-account';
import { MatSnackBar } from '@angular/material/snack-bar';
import { AccountService } from '../../services/account.service';
import { Router } from '@angular/router';
var sha256 = require('js-sha256');
@Component({
  selector: 'app-register',
  templateUrl: './register.component.html',
  styleUrls: ['./register.component.css'],
})
export class RegisterComponent implements OnInit {
  Users: CreateAccount[] = [];
  newAccount: CreateAccount = new CreateAccount();
  age: number = 0;
  checkboxStatus: Boolean = false;
  constructor(
    private service: AccountService,
    private _snackBar: MatSnackBar,
    private router: Router
  ) {}

  /**
   * Este metodo permite realizar un pequeño delay
   * @param ms el tiempo del delay en ms
   */
  async delay(ms: number) {
    await new Promise<void>((resolve) => setTimeout(() => resolve(), ms)).then(
      () => console.log('fired')
    );
  }

  ngOnInit(): void {}

  CData: any[] = countriesData;

  /**
   * Metodo para mostrar alerta de error por 2 segundos
   * @param message1 Mensaje de error
   * @param message2 Mensaje para cerrar alerta
   */
  openError(message: string, message2: string) {
    this._snackBar.open(message, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'red-snackbar',
    });
  }

  /**
   * Metodo para mostrar alerta de exito por 2 segundos
   * @param message1 Mensaje de exito
   * @param message2 Mensaje para cerrar alerta
   */
  openSuccess(message1: string, message2: string) {
    this._snackBar.open(message1, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'green-snackbar',
    });
  }

  /**
   * Metodo para calcular la edad a partir de la fecha de nacimiento
   * digitado por el usuario
   */
  CalculateAge() {
    if (this.newAccount.birthdate) {
      var timeDiff = Math.abs(
        Date.now() - new Date(this.newAccount.birthdate).getTime()
      );
      this.age = Math.floor(timeDiff / (1000 * 3600 * 24) / 365.25);

      console.log(this.age);
    }
  }

  /**
   * Método para verificar fomrato de correo utilizando estándar RFC 2822
   * @param email
   * @returns Booleano de validez de correo
   */
  checkEmailFormat(email: string) {
    let sampleRegExMail = new RegExp(
      "[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?"
    );
    return sampleRegExMail.test(email);
  }

  /**
   * Método para obtener estado de checkbox para TyC
   */
  changeStatus() {
    this.checkboxStatus = !this.checkboxStatus;
  }

  /**
   * Este metodo permite realizar las verificaciones del formulario de crear
   * cuenta asi como posteriormente realizar el envio de los datos.
   */
  addUser() {
    this.delay(50).then(() => {
      this.service
        .getAccountByEmail(this.newAccount.email)
        .subscribe((data) => (this.Users = data));
    });

    //Verificar que esten todos los espacios requeridos
    if (
      this.newAccount.email == '' ||
      this.newAccount.password == '' ||
      this.newAccount.name == '' ||
      this.newAccount.lastname == '' ||
      this.newAccount.nickname == '' ||
      this.newAccount.birthdate == '' ||
      this.newAccount.country == ''
    ) {
      this.openError(
        'Faltan espacios requeridos para registrarse',
        'Intente de nuevo'
      );
    } else {
      this.CalculateAge();

      if (!this.checkEmailFormat(this.newAccount.email)) {
        this.openError('Formato de correo inválido', 'Intente de nuevo');
      } else if (this.Users.length > 0) {
        this.openError(
          'Hay una cuenta existente con este correo',
          'Ingrese otra para continuar'
        );
      } else if (
        !(
          this.newAccount.password.length >= 6 &&
          this.newAccount.password.length <= 8
        )
      ) {
        this.openError(
          'La contraseña debe tener una extensión entre 6 y 8 caractéres',
          'Intente de nuevo'
        );
      } else if (!(this.age >= 18)) {
        this.openError('Debe ser mayor de 18 años', 'Intente de nuevo');
      } else if (!this.checkboxStatus) {
        this.openError('Debe aceptar términos y condiciones', 'Volver');
      } else {
        this.newAccount.password = sha256(this.newAccount.password);
        this.service
          .createAccount(this.newAccount)
          .subscribe((data) => console.log(data));
        this.openSuccess('Cuenta creada con éxito', 'Ok');
        this.router.navigate(['/login']);
      }
    }
  }
}
