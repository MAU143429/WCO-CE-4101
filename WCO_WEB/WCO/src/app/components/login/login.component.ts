import { Component, OnInit } from '@angular/core';
import { NgbDateNativeAdapter } from '@ng-bootstrap/ng-bootstrap';
import { Login } from 'src/app/model/login';
import { CreateAccount } from '../../model/create-account';
import { AccountService } from '../../services/account.service';
import {MatSnackBar} from '@angular/material/snack-bar';
import { Router } from '@angular/router';


@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.css'],
})
export class LoginComponent implements OnInit {
  Users: CreateAccount[] = [];
  newLogin: Login = new Login();

  /**
   * Este metodo permite realizar un pequeño delay
   * @param ms el tiempo del delay en ms
   */
   async delay(ms: number) {
    await new Promise<void>((resolve) => setTimeout(() => resolve(), ms)).then(
      () => console.log('fired')
    );
  }  

  constructor(
    private service: AccountService,
    private _snackBar: MatSnackBar,
    private router: Router
    ) {}

  ngOnInit(): void {}

  /**
   * Metodo para mostrar alerta de error por 2 segundos
   * @param message1 Mensaje de error
   * @param message2 Mensaje para cerrar alerta
   */
  openError(message: string) {
    this._snackBar.open(message, 'Intente de nuevo',{
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'red-snackbar',      
    })
  };

  /**
   * Metodo para mostrar alerta de exito por 2 segundos
   * @param message1 Mensaje de exito
   * @param message2 Mensaje para cerrar alerta
   */
  openSuccess(message1: string, message2: string) {
    this._snackBar.open(message1, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'green-snackbar',
    });
  }


  /**
   * Este metodo permite hacer todas las verificaciones del inicio de sesion
   * asi como la comprobacion de contraseñas y posteriormente el paso a WCO
   */
  addLogin() {

    this.delay(50).then(() => {
      this.service
        .getAccountByEmail(this.newLogin.email)
        .subscribe((data) => (this.Users = data));
    });

    //Verificar que esten todos los espacios requeridos
    if ((this.newLogin.email == "" ||
         this.newLogin.password == "")){
      this.openError("Faltan espacios requeridos para iniciar sesión")
    }
    // Verificar es que el correo tenga una cuenta en WCO (Si verifico esto, y despues digo que credenciales estan mal, entonces es obvio que la contraseña es la que está mal)
    else if (!(this.Users.length > 0)){
      this.openError("Cuenta con correo no registrado")
    }
    // Verificar las credenciales
    else if (!(this.Users[0].password == this.newLogin.password)){
      this.openError("Credenciales incorrectos")
    } else {
      this.openSuccess('Cuenta creada con éxito', 'Ok');
      this.router.navigate(['/home']);
    }
  }
}
