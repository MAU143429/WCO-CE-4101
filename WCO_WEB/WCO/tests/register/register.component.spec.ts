import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
import { RegisterComponent } from '../../src/app/components/register/register.component';

import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing'
import { AccountService } from '../../src/app/services/account.service';
import { MatSnackBar, MatSnackBarModule } from '@angular/material/snack-bar';
import { Observable, observable, of } from 'rxjs';
import { Router } from '@angular/router';
import { FormsModule, NgModel, NgSelectOption, ReactiveFormsModule } from '@angular/forms';
import { EventManager } from '@angular/platform-browser';
import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { LoginComponent } from '../../src/app/components/login/login.component';


const dummyAccount = [{
  name: 'David',
  lastname: 'Soto',
  email: 'prueba@gmail.com',
  password: 'lunes233',
  nickname: 'dasotovid',
  birthdate: '01/01/2001',
  country: 'Chile'
}];

const mockAccountService: {
  createAccount: () => Observable<any>,
  getAccountByEmail: () => Observable<any>,
  getAccountByNickname: () => Observable<any>
} = {
  createAccount: () => of(dummyAccount),
  getAccountByEmail: () => of([]),
  getAccountByNickname: () => of([])
};


describe('RegisterComponent', () => {
  let component: RegisterComponent;
  let fixture: ComponentFixture<RegisterComponent>;
  let compiled: HTMLElement;
  let service: AccountService;
  let httpMock: HttpTestingController;
  let router: Router;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ RegisterComponent ],
      imports: [ 
        HttpClientTestingModule, 
        MatSnackBarModule,
        //FormsModule, 
        //ReactiveFormsModule,
        
        BrowserAnimationsModule,
        NoopAnimationsModule,
        RouterTestingModule.withRoutes([{path: 'login', component: LoginComponent}])
        ],
      providers: [ 
        { provide: AccountService, useValue: mockAccountService }
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(RegisterComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( AccountService );
    httpMock = TestBed.inject( HttpTestingController );
    router = TestBed.inject(Router);

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test('Valid email', () => {
    let test = component.checkEmailFormat("usuario@gmail.com")
    expect(test).toBeTruthy();
  })

  test('Invalid email', () => {
    let test = component.checkEmailFormat("usuario2gmail.com")
    expect(test).toBeFalsy();
  })

  test('Valid password', () => {
    let test = component.checkPasswordFormat("AA8POO")
    expect(test).toBeTruthy();
  })

  test('Invalid password', () => {
    let test = component.checkPasswordFormat("121456")
    expect(test).toBeFalsy();
  })

  test('Age 22 true', () => {
    component.newAccount.birthdate = '06/06/2000'
    component.CalculateAge();
    expect(component.age).toBe(22);
  })

  test('Age 19 false', () => {
    component.newAccount.birthdate = '06/06/2000'
    component.CalculateAge();
    expect(component.age == 19).toBeFalsy();
  })

  test('Change checkbox status to true', () => {
    component.changeStatus();
    expect(component.checkboxStatus).toBeTruthy();
  })

  test('Change checkbox status to false', () => {
    component.changeStatus();
    component.changeStatus();
    expect(component.checkboxStatus).toBeFalsy();
  })

  test('se prueba addUser()', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const routeSpy = jest.spyOn(router, 'navigate');

    component.newAccount.email = 'prueba@gmail.com'
    component.newAccount.password = 'lunes233'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2001"
    component.newAccount.country = "Chile"
    component.checkboxStatus = true;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(routeSpy).toHaveBeenCalledWith(["/login"]);
  }))


  test('se prueba addUser() sin ingresar correo', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const errorSpy = jest.spyOn(component, 'openError');

    component.newAccount.email = ''
    component.newAccount.password = 'lunes233'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2001"
    component.newAccount.country = "Chile"
    component.checkboxStatus = true;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(errorSpy).toHaveBeenCalledWith("Faltan espacios requeridos para registrarse", "Intente de nuevo");
  }))

  test('se prueba addUser() sin aceptar TyC', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const errorSpy = jest.spyOn(component, 'openError');

    component.newAccount.email = 'prueba@gmail.com'
    component.newAccount.password = 'lunes233'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2001"
    component.newAccount.country = "Chile"
    component.checkboxStatus = false;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(errorSpy).toHaveBeenCalledWith("Debe aceptar términos y condiciones", "Volver");
  }))

  test('se prueba addUser() con formato incorrecto de contraseña', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const errorSpy = jest.spyOn(component, 'openError');

    component.newAccount.email = 'prueba@gmail.com'
    component.newAccount.password = 'lunes'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2001"
    component.newAccount.country = "Chile"
    component.checkboxStatus = true;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(errorSpy).toHaveBeenCalledWith('Formato de contraseña inválido, debe ser alfanumérica',
    'Intente de nuevo');
  }))

  test('se prueba addUser() con formato de contraseña eróoneo', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const errorSpy = jest.spyOn(component, 'openError');

    component.newAccount.email = 'prueba@gmail.com'
    component.newAccount.password = 'pt5'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2001"
    component.newAccount.country = "Chile"
    component.checkboxStatus = true;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(errorSpy).toHaveBeenCalledWith('La contraseña debe tener una extensión entre 6 y 8 caractéres',
    'Intente de nuevo');
  }))


  test('se prueba addUser() con menor de 18 años', fakeAsync(() => {
    const getAccESpy = jest.spyOn(service, "getAccountByEmail");
    getAccESpy.mockReturnValue(of([]));

    const getAccNSpy = jest.spyOn(service, "getAccountByNickname");
    getAccNSpy.mockReturnValue(of([]));

    const createAccSpy = jest.spyOn(service, 'createAccount');
    createAccSpy.mockReturnValue(of(dummyAccount));

    const errorSpy = jest.spyOn(component, 'openError');

    component.newAccount.email = 'prueba@gmail.com'
    component.newAccount.password = 'lunes223'
    component.newAccount.name = 'David'
    component.newAccount.lastname = 'Soto'
    component.newAccount.nickname = 'dasotovid'
    component.newAccount.birthdate = "01/01/2011"
    component.newAccount.country = "Chile"
    component.checkboxStatus = true;

    fixture.detectChanges();

    component.addUser();
    tick(300);

    expect(errorSpy).toHaveBeenCalledWith('Debe ser mayor de 18 años', 'Intente de nuevo');
  }))


});
