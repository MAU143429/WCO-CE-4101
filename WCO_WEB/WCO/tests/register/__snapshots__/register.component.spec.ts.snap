// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterComponent debe de hace match con el snapshot 1`] = `"<html lang=\\"en\\" dir=\\"ltr\\" ng-reflect-dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><body><div class=\\"container\\"><div class=\\"title\\">Crear una cuenta</div><div class=\\"content\\"><form><div class=\\"account-details\\"><div class=\\"input-box\\"><span class=\\"details\\">Nombre<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"text\\" maxlength=\\"30\\" placeholder=\\"Ingrese un nombre aquí...\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Apellidos<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"text\\" maxlength=\\"30\\" placeholder=\\"Ingrese los apellidos aquí...\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Nickname<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"text\\" maxlength=\\"30\\" placeholder=\\"Ingrese un nickname...\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Correo Electrónico<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"email\\" maxlength=\\"30\\" placeholder=\\"Ingrese un correo electrónico..\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Contraseña<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"password\\" minlength=\\"6\\" maxlength=\\"8\\" placeholder=\\"Ingrese una contraseña aquí...\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Fecha de nacimiento<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"date\\" required=\\"\\"></div><div class=\\"input-box1\\"><span class=\\"details\\">País de Procedencia<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><mbsc-select placeholder=\\"Seleccione su país de procedencia\\"></mbsc-select></div><div class=\\"input-box2\\"><span style=\\"color: rgb(4, 0, 255);\\"><a href=\\"https://www.fifa.com/es/terms-of-service\\">Mostrar Términos y Condiciones</a></span><br><span><input type=\\"checkbox\\"> Aceptar Términos y Condiciones</span></div><div class=\\"input-box2\\"><span>Ya tienes una cuenta? Ingresa <a routerlink=\\"/login\\" style=\\"color: rgb(4, 0, 255);\\" ng-reflect-router-link=\\"/login\\" href=\\"/login\\">aquí</a></span></div></div><div class=\\"button\\"><input type=\\"submit\\" value=\\"Crear cuenta\\"></div></form></div></div></body></html>"`;
